gcloud/storage/api.py

```diff
@@ -184,6 +170,11 @@ class _BucketIterator(Iterator):
     """
 
     def __init__(self, connection, extra_params=None):
+        # If an implicit connection was intended, we pass along `None` to the
+        # Bucket() constructor as well.
```

Comment:
This bit doesn't seem correct:  why make the `Bucket` instances do the lookup again?